cli/test/fetchPausePlans.test.js

```diff
@@ -0,0 +1,272 @@
+import { jest, describe, it, expect, beforeAll } from "@jest/globals";
+import { ethers } from "ethers";
+import {
+  fetchPausePlans,
+  decodeLogNote,
+  decodeCallParams,
+  hash,
+  processEvent,
+} from "../fetchPausePlans.js";
+import pauseABI from "../pause_abi.json" with { type: "json" };
+import { mockEvents } from "./fixtures/events.js";
+
+const PLOT_TOPIC =
+  "0x46d2fbbb00000000000000000000000000000000000000000000000000000000";
+const EXEC_TOPIC =
+  "0x168ccd6700000000000000000000000000000000000000000000000000000000";
+const DROP_TOPIC =
+  "0x162c7de300000000000000000000000000000000000000000000000000000000";
+
+let eventStats;
+let processedEvents;
+
+function analyzeEvents(events) {
+  const plotEvents = events.filter((e) => e.topics[0] === PLOT_TOPIC);
+  const execEvents = events.filter((e) => e.topics[0] === EXEC_TOPIC);
+  const dropEvents = events.filter((e) => e.topics[0] === DROP_TOPIC);
+
+  return {
+    total: events.length,
+    plotCount: plotEvents.length,
+    execCount: execEvents.length,
+    dropCount: dropEvents.length,
+    pendingCount: plotEvents.length - (execEvents.length + dropEvents.length),
+    events: {
+      plot: plotEvents,
+      exec: execEvents,
+      drop: dropEvents,
+    },
+  };
+}
+
+describe("fetchPausePlans", () => {
+  beforeAll(() => {
+    const mockContract = {
+      interface: new ethers.Interface(pauseABI),
+    };
+    processedEvents = mockEvents.map((event) =>
+      processEvent(event, mockContract),
+    );
```

Comment:
Initially, I was a bit confused with the `mockContract` vs `mockContractInstance` variables, but I get now that the former is only used to process the fixtures.

I'd suggest to not create a variable for it and just pass it inline, as it's the only time it's used.
```suggestion
    processedEvents = mockEvents.map((event) =>
      processEvent(event, {
        interface: new ethers.Interface(pauseABI),
      }),
    );
```

Reply:
done in [c6d9fcf](https://github.com/makerdao/protego/pull/8/commits/c6d9fcfd4930b961f5a3cf80baa2177ad0d510e2)